spdata/reader.py

```diff
@@ -45,3 +45,26 @@ def load_txt(content) -> ty.Dataset:
     coordinates = ty.Coordinates(*zip(*coordinates))
 
     return ty.Dataset(data, coordinates, mzs, labels)
+
+def load_imzml(file_path) -> ty.Dataset:
+    """Load Dataset from imzml file
+
+    Args:
+        file_path: path to imzml file
+
+    Returns:
+        out: spdata.types.Dataset
+    """
+    with imzparse(file_path) as input_handle:
+        assert np.min(input_handle.mzLengths) == np.max(input_handle.mzLengths)
+        mzs = input_handle.getspectrum(0)[0]
+        spectra = [
+            input_handle.getspectrum(i)[1]
+            for i in range(len(input_handle.coordinates))
```

Comment:
We do not know, how this `input_handle.coordinates` behaves. I'd rather get all coordinates first, just to use the length of parsed structure. We do not want to iterate `.imzML` metadata twice.

Reply:
👍